roiextractors/extractors/nwbextractors/nwbextractors.py

```
@@ -180,82 +180,65 @@ def __init__(self, file_path):
-        # self.image_masks = np.moveaxis(np.array(ps['image_mask'].data), [0, 1, 2], [2, 0, 1])
-        if 'image_mask' in ps.colnames:
-            self.image_masks = DatasetView(ps['image_mask'].data).lazy_transpose([1, 2, 0])
-        if 'RoiCentroid' in ps.colnames:
-            self._roi_locs = ps['RoiCentroid']
-        if 'Accepted' in ps.colnames:
-            self._accepted_list = ps['Accepted'].data[:]
-        # Extract Image dimensions:
-
-        # Extract roi_response:
-        roi_names = [_nwbchildren_name[val]
-                      for val, i in enumerate(_nwbchildren_type) if i == 'RoiResponseSeries']
-        if not roi_names:
-            raise Exception('no ROI response series found')
+        self.nwbfile = self._io.read()
+        self._raw_movie_file_location = None
+
+        ophys = self.nwbfile.processing.get('ophys')
+        if ophys is None:
+            warn('could not find ophys processing module in nwbfile')
         else:
-            for trace_name in ['roiresponseseries', 'dff', 'neuropil', 'deconvolved']:
-                trace_name_nwb = [j for j,i in enumerate(roi_names) if trace_name in i.lower()]
-                if trace_name_nwb:
-                    trace_name_segext = 'raw' if trace_name == 'roiresponseseries' else trace_name
-                    setattr(self,f'_roi_response_{trace_name_segext}',
-                            mod['Fluorescence'].get_roi_response_series(roi_names[trace_name_nwb[0]]).data[:].T)
-
-        # Extract samp_freq:
-        self._sampling_frequency = mod['Fluorescence'].get_roi_response_series(roi_names[0]).rate
-        # Extract get_num_rois()/ids:
-        self._roi_idx = np.array(ps.id.data)
+            # Extract roi_response:
+            if 'Fluorescence' in ophys.data_interfaces:
+                fluorescence = ophys.data_interfaces['Fluorescence']
+            else:
+                raise Exception('could not find Fluorescence module in nwbfile')
```

Fluorescence is not a module.

I find it strange that no ophys ProcessingModule is a warning while no Fluorescence object is an exception. What is the logic behind this?

---

yes, it should throw an error at processing not found itself: fixed https://github.com/catalystneuro/roiextractors/pull/46/commits/33b230ae19626a880d3e70a62c318905c132f6ff